examples/jetstream/limits-stream/java/Main.java

```diff
@@ -0,0 +1,135 @@
+            // be explained in later examples.
+            //
+            // JetStream provides both file and in-memory storage options. For
+            // durability of the stream data, file storage must be chosen to
+            // survive crashes and restarts. This is the default for the stream,
+            // but we can still set it explicitly.
+            String streamName = "EVENTS";
+            StreamConfiguration.Builder streamConfigBuilder = StreamConfiguration.builder()
+                    .name(streamName)
+                    .subjects("events.>")
+                    .storageType(StorageType.File);
+
+            // Finally, let's add/create the stream with the default (no) limits.
+            StreamInfo stream = jsm.addStream(streamConfigBuilder.build());
+            System.out.println("Created the stream.");
+
+            // Let's publish a few messages which are received by the stream since
+            // they match the subject bound to the stream. The `js.publish` method
+            // is a convenience for sending a `nc.request` and waiting for the
+            // acknowledgement.
+            js.publish("events.page_loaded", null);
+            js.publish("events.mouse_clicked", null);
+            js.publish("events.mouse_clicked", null);
+            js.publish("events.page_loaded", null);
+            js.publish("events.mouse_clicked", null);
+            js.publish("events.input_focused", null);
+            System.out.println("Published 6 messages.");
+
+            // There is also an async form in which the client batches the
+            // messages to the server and then asynchronously receives
+            // the acknowledgements.
+            List<CompletableFuture<PublishAck>> futures = new ArrayList<>();
+            futures.add(js.publishAsync("events.input_changed", null));
+            futures.add(js.publishAsync("events.input_blurred", null));
+            futures.add(js.publishAsync("events.key_pressed", null));
+            futures.add(js.publishAsync("events.input_focused", null));
+            futures.add(js.publishAsync("events.input_changed", null));
+            futures.add(js.publishAsync("events.input_blurred", null));
+
+            CompletableFuture<Void> publishCompleted = CompletableFuture.allOf(futures.toArray(new CompletableFuture[0]));
+            try {
+                publishCompleted.join();
+                System.out.println("Published 6 messages.");
+            } catch (Exception e) {
+                System.out.println("Publish failed " + e);
+            }
+
+            // Checking out the stream info, we can see how many messages we
+            // have.
+            printStreamState(jsm, streamName);
+
+            // Stream configuration can be dynamically changed. For example,
+            // we can set the max messages limit to 10 and it will truncate the
+            // two initial events in the stream.
```

Comment:
Much stream configuration can be changed (updated) but not all configuration is allowed to be changed.

See tests:

* testUpdateStream line 182
* testUpdateStreamInvalids line 303

in here:
https://github.com/nats-io/nats.java/blob/main/src/test/java/io/nats/client/impl/JetStreamManagementTests.java

Reply:
Have added the second paragraph.
```java
// Stream configuration can be dynamically changed. For example,
// we can set the max messages limit to 10 and it will truncate the
// two initial events in the stream.
//
// Most fields can be edited, but some are not allowed to be changed.
// See also: https://docs.nats.io/nats-concepts/jetstream/streams#configuration.
```